corehq/form_processor/tests/test_gdpr_scrub_user_from_forms.py

```diff
@@ -0,0 +1,118 @@
+<data uiVersion="1" version="17" name="New Form" xmlns:jrm="http://dev.commcarehq.org/jr/xforms"
+    xmlns="http://openrosa.org/formdesigner/form-processor">
+    <dalmation_count>yes</dalmation_count>
+    <n0:meta xmlns:n0="http://openrosa.org/jr/xforms">
+        <n0:deviceID>DEV IL</n0:deviceID>
+        <n0:timeStart>2013-04-19T16:53:02.000000Z</n0:timeStart>
+        <n0:timeEnd>2013-04-19T16:52:02.000000Z</n0:timeEnd>
+        <n0:username>replacement_username</n0:username>
+        <n0:userID>cruella_deville</n0:userID>
+        <n1:appVersion xmlns:n1="http://commcarehq.org/xforms"></n1:appVersion>
+    </n0:meta>
+</data>"""
+
+NEW_USERNAME = "replacement_username"
+
+
+class UpdateFormTests(TestCase):
+
+    def test_update_form_data(self):
+        form = get_simple_wrapped_form(uuid.uuid4().hex,
+                                       metadata=TestFormMetadata(domain=DOMAIN),
+                                       simple_form=GDPR_SIMPLE_FORM)
+        actual_form_xml = Command().update_form_data(form, NEW_USERNAME)
+        self.assertXMLEqual(EXPECTED_FORM_XML, actual_form_xml)
+
+@use_sql_backend
+class GDPRScrubUserFromFormsSqlTests(TestCase):
+    def setUp(self):
+        super(GDPRScrubUserFromFormsSqlTests, self).setUp()
+        self.db = TemporaryFilesystemBlobDB()
+
+    def tearDown(self):
+        self.db.close()
+        super(GDPRScrubUserFromFormsSqlTests, self).tearDown()
+
+    def test_modify_attachment_xml_and_metadata_sql(self):
+        form = get_simple_wrapped_form(uuid.uuid4().hex,
+                                       metadata=TestFormMetadata(domain=DOMAIN),
+                                       simple_form=GDPR_SIMPLE_FORM)
+
+        new_form_xml = Command().update_form_data(form, NEW_USERNAME)
+
+        FormAccessors(DOMAIN).modify_attachment_xml_and_metadata(form, new_form_xml)
+
+        # Test that the xml changed
+        actual_form_xml = form.get_attachment("form.xml")
+        self.assertXMLEqual(EXPECTED_FORM_XML, actual_form_xml)
+
+        # Test that the metadata changed in the database
+        attachment_metadata = form.get_attachment_meta("form.xml")
+        actual_form_xml_from_db = XFormAttachmentSQL.read_content(attachment_metadata)
+        self.assertXMLEqual(EXPECTED_FORM_XML, actual_form_xml_from_db)
+
+        # Test that the operations history is updated in this form
+        refetched_form = FormAccessors(DOMAIN).get_form(form.form_id)
+        self.assertEqual(len(refetched_form.history), 1)
+        self.assertEqual(refetched_form.history[0].operation, "Scrub username for GDPR compliance.")
+
+
+class GDPRScrubUserFromFormsCouchTests(TestCase):
+    def setUp(self):
+        super(GDPRScrubUserFromFormsCouchTests, self).setUp()
+        self.db = TemporaryFilesystemBlobDB()
+
+    def tearDown(self):
+        self.db.close()
+        super(GDPRScrubUserFromFormsCouchTests, self).tearDown()
+
+    def test_modify_attachment_xml_and_metadata_couch(self):
```

Comment:
These tests look the same. The common pattern is to do this:

```
class CouchTests(TestCase):
    def my_test(self):
        ....

@use_sql_backend
class SQLtests(CouchTests)
    pass


```

Reply:
cool, didn't know I could do that, fixed here:

https://github.com/dimagi/commcare-hq/pull/19979/commits/34a94c2a68078a5d77e1bbbaad68465d5120f2d9